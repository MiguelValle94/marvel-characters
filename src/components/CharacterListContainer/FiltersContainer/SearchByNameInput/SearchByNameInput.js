import React from 'react';
import PropTypes from 'prop-types';

import './SearchByNameInput.scss'

export default function SearchByNameInput({ onChange }) {
  function handleOnChange(event) {
    const { length } = event.target.value;
    if (length === 0 || length >= 3) {
      onChange(event);
    }
  }

  return (
    <div className='input-group'>
      <input
        className='form-control border-right-0 border'
        type='text'
        name='nameStartsWith'
        autoComplete='off'
        placeholder='Search name'
        onChange={(e) => handleOnChange(e)}
      />
      <span className='input-group-append'>
        <div className='input-group-text'>
          <i className='fa fa-search'></i>
        </div>
      </span>
    </div>
  );
}

SearchByNameInput.propTypes = {
  onChange: PropTypes.func.isRequired
};
