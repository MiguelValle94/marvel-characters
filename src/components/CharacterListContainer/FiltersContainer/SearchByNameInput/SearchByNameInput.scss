@import '../../../../styles/variables.scss';

.input-group {
  margin-bottom: var(--space);
  width: auto;
}
