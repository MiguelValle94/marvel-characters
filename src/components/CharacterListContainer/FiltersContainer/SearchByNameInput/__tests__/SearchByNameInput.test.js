import React from 'react';
import { mount } from 'enzyme';

import SearchByNameInput from '../SearchByNameInput';

const setUpMount = (value, handleOnChange) => {
  const component = mount(
    <SearchByNameInput value={value} onChange={handleOnChange} />
  );
  return component;
};

describe('SearchByNameInput', () => {
  const handleOnChange = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render component', () => {
    const component = setUpMount('', handleOnChange);
    expect(component).toMatchSnapshot();
  });

  it('should call onChange callback on input change', () => {
    const name = 'nameStartsWith';
    const value = 'spi';
    const event = { target: { value, name } };
    const component = setUpMount('name', handleOnChange);
    const input = component.find('input');

    expect(handleOnChange).not.toHaveBeenCalled();
    input.simulate('change', event);

    expect(handleOnChange).toHaveBeenCalledTimes(1);
  });

  it('should call onChange callback on input change if length === 0', () => {
    const name = 'nameStartsWith';
    const value = '';
    const event = { target: { value, name } };
    const component = setUpMount('name', handleOnChange);
    const input = component.find('input');

    expect(handleOnChange).not.toHaveBeenCalled();
    input.simulate('change', event);

    expect(handleOnChange).toHaveBeenCalledTimes(1);
  });

  it('should not call onChange callback on input change if 3 > length > 0', () => {
    const name = 'nameStartsWith';
    const value = 'sp';
    const event = { target: { value, name } };
    const component = setUpMount('name', handleOnChange);
    const input = component.find('input');

    expect(handleOnChange).not.toHaveBeenCalled();
    input.simulate('change', event);

    expect(handleOnChange).not.toHaveBeenCalled();
  });
});
