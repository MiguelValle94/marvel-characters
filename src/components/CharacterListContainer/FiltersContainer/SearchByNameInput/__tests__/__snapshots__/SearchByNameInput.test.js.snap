// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchByNameInput should render component 1`] = `
<SearchByNameInput
  onChange={[MockFunction]}
  value=""
>
  <div
    className="input-group"
  >
    <input
      autoComplete="off"
      className="form-control border-right-0 border"
      name="nameStartsWith"
      onChange={[Function]}
      placeholder="Search name"
      type="text"
    />
    <span
      className="input-group-append"
    >
      <div
        className="input-group-text"
      >
        <i
          className="fa fa-search"
        />
      </div>
    </span>
  </div>
</SearchByNameInput>
`;
